backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java

```diff
@@ -0,0 +1,138 @@
+package se.chalmers.dat265.group1.api.order;
+
+import se.chalmers.dat265.group1.api.ApiController;
+import se.chalmers.dat265.group1.model.Order;
+import se.chalmers.dat265.group1.model.dbEntities.OrderData;
+import se.chalmers.dat265.group1.model.dbEntities.OrderedPart;
+
+import java.text.SimpleDateFormat;
+import java.util.LinkedList;
+import java.util.List;
+
+public class OrderController extends ApiController implements OrderAPI {
+    public OrderController(boolean debug) {
+        super(debug);
+    }
+
+    private static final String DATE_FORMAT = "yyyy-MM-dd";
+
+    @Override
+    public List<Order> getAllOrders() {
+        List<OrderData> orderDataList = orderDataRepository.getObjects();
+        return getMerged(orderDataList);
+
+    }
+
+    /**
+     * At the moments only returns the specific orderPart and not all the ordered parts in an order
+     *
+     * @param digitalPartID filter
+     * @return orders with specific with specific part
+     */
+    @Override
+    public List<Order> getAllOrdersWithDigitalPart(String digitalPartID) {
+        List<OrderedPart> orderedPartList = orderedPartRepository.getObjects("digitalPartID=" + digitalPartID);
+        List<Order> orders = new LinkedList<>();
+        for (OrderedPart orderedPart : orderedPartList) {
+            OrderData tempData = orderDataRepository.getObject(orderedPart.getDigitalPartID());
+            boolean alreadyFound = false;
+            for (Order order : orders) {
+                if (tempData.getId() == order.getId()) {
+                    alreadyFound = true;
+                    break;
+                }
+            }
+            if (!alreadyFound) {
+                List<OrderedPart> singleList = new LinkedList<>();
+                singleList.add(orderedPart);
+                Order order = new Order(tempData, singleList);
+                orders.add(order);
+            }
+        }
+        return orders;
+    }
+
+    private List<Order> getMerged(List<OrderData> orderDataList) {
+        List<Order> orders = new LinkedList<>();
+        for (OrderData orderData : orderDataList) {
+            Order temp = new Order(orderData, orderedPartRepository.getObjects("orderID=" + orderData.getId()));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "orderID=" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)